### PR TITLE
First Revision of TTS Docs and Notebooks Update for 1.0

### PR DESCRIPTION
**Pull request by blisc**

### CODE REVIEW

File: docs/source/tts/intro.rst [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+   datasets
+   api
+
 Speech Synthesis or Text-to-Speech (TTS) involves turning text into human speech. The NeMo TTS collection currently
-supports a two stage pipeline. First, a model is used to generate a mel spectrogram from text. Second, a model is used
+supports two pipelines for TTS:
+
+1) The two stage pipeline. First, a model is used to generate a mel spectrogram from text. Second, a model is used
 to generate audio from a mel spectrogram.
+2) The "end-to-end" approach that uses one model to generate audio straight from speech.
```
Comment (redoctopus) [inline docs/source/tts/intro.rst:16]: Straight from text rather than speech?

File: docs/source/tts/intro.rst [hunk 2]
```diff
@@ -63,7 +76,15 @@ NeMo supports a variety of models that can be used for TTS.
    * - FastSpeech 2
      - :class:`SpectrogramGenerator<nemo.collections.tts.models.base.SpectrogramGenerator>`
      - (Coming soon)
-     - Non-autoregressive spectrogram generator that predicts duration, energy, and pitch
+     - Non-autoregressive transformer-based spectrogram generator that predicts duration, energy, and pitch
+   * - FastPitch
+     - :class:`SpectrogramGenerator<nemo.collections.tts.models.base.SpectrogramGenerator>`
+     - (Coming soon)
+     - Non-autoregressive transformer-based spectrogram generator that predicts duration, and pitch
```
Comment (redoctopus) [inline docs/source/tts/intro.rst:83]: Remove comma

File: docs/source/tts/intro.rst [hunk 3]
```diff
@@ -84,32 +105,48 @@ NeMo supports a variety of models that can be used for TTS.
...
+*******************
+
+Correspondingly, NeMo TTS has one base class for the end-to-end pipeline:
+
+  - :class:`TextToWaveform<nemo.collections.tts.models.base.TextToWaveform>`
 
-.. autoclass:: nemo.collections.tts.models.base.Vocoder
-    :show-inheritance:
-    :members:
+Similiary to the SpectrogramGenerator, :class:`TextToWaveform<nemo.collections.tts.models.base.TextToWaveform>`
```
Comment (redoctopus) [inline docs/source/tts/intro.rst:145]: Minor typo: s/Similiary/Similarly

File: docs/source/tts/intro.rst [hunk 4]
```diff
@@ -84,32 +105,48 @@ NeMo supports a variety of models that can be used for TTS.
...
+implements two functions: :py:meth:`parse<nemo.collections.tts.models.base.TextToWaveform.parse>` which
+accepts raw python strings and returns a torch.tensor that represents tokenized text ready to pass to, and
+:py:meth:`generate_spectrogram<nemo.collections.tts.models.base.TextToWaveform.convert_text_to_waveform>` which
+accepts a batch of tokenized text and returns a torch.tensor that represents a batch of audio,
```
Comment (redoctopus) [inline docs/source/tts/intro.rst:149]: Period instead of comma?

File: tutorials/tts/1_TTS_inference.ipynb [hunk 5]
```diff
@@ -83,30 +89,42 @@
...
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "e2e_model = e2e_selector.value\n",
+    "spectrogram_generator = spectrogram_generator_selector.value\n",
+    "audio_generator = audio_generator_selector.value\n",
+    "\n",
+    "if e2e_model is None and spectrogram_generator is None and audio_generator is None:\n",
+    "    raise ValueError(\"No models were chosed. Please return to the previous step and choose either 1 end-to-end model or 1 spectrogram generator and 1 vocoder.\")\n",
```
Comment (redoctopus) [inline tutorials/tts/1_TTS_inference.ipynb:117]: s/chosed/chosen

File: docs/source/tts/intro.rst (hunk 1 above)
Comment (blisc) [reply docs/source/tts/intro.rst]: Done

File: docs/source/tts/intro.rst (hunk 2 above)
Comment (blisc) [reply docs/source/tts/intro.rst]: Done

File: docs/source/tts/intro.rst (hunk 3 above)
Comment (blisc) [reply docs/source/tts/intro.rst]: Done

File: docs/source/tts/intro.rst (hunk 4 above)
Comment (blisc) [reply docs/source/tts/intro.rst]: Done

File: tutorials/tts/1_TTS_inference.ipynb (hunk 5 above)
Comment (blisc) [reply tutorials/tts/1_TTS_inference.ipynb]: Done

**Review by redoctopus — APPROVED** [review]
LGTM

**Pull request merged**